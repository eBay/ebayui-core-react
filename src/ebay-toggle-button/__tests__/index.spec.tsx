import React from 'react'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import EbayToggleButton from '../index'
import { composeStories } from '@storybook/react'
import * as stories from './index.stories'

const {
    Default,
    WithChildren,
    IconButton,
    ImageButton,
    ImageButtonWithPlacement
} = composeStories(stories)

describe('<EbayToggleButton />', () => {
    it('should render component - <Default />', () => {
        const { asFragment } = render(<Default />)

        const snapshot = asFragment()
        expect(snapshot).toMatchSnapshot()
    })
    it('should render component - <WithChildren />', () => {
        const { asFragment } = render(<WithChildren />)

        const snapshot = asFragment()
        expect(snapshot).toMatchSnapshot()
    })
    it('should render component - <IconButton />', () => {
        const { asFragment } = render(<IconButton />)

        const snapshot = asFragment()
        expect(snapshot).toMatchSnapshot()
    })
    it('should render component - <ImageButton />', () => {
        const { asFragment } = render(<ImageButton />)

        const snapshot = asFragment()
        expect(snapshot).toMatchSnapshot()
    })
    it('should render component - <ImageButtonWithPlacement />', () => {
        const { asFragment } = render(<ImageButtonWithPlacement />)

        const snapshot = asFragment()
        expect(snapshot).toMatchSnapshot()
    })
    it('should fire onToggle when clicking', async () => {
        const mockOnToggle = jest.fn()

        const { getByRole } = render(<Default onToggle={mockOnToggle} />)

        const button = getByRole('button', { name: /title/i })
        await userEvent.click(button)

        expect(mockOnToggle).toHaveBeenCalledTimes(1)
    })
    it('should not click if disabled', async () => {
        const mockOnToggle = jest.fn()

        const { getByRole } = render(<Default disabled={true} />)

        const button = getByRole('button', { name: /title/i })
        expect(button).toBeDisabled()

        await userEvent.click(button)
        // Assert that the onToggle callback was not called
        expect(mockOnToggle).not.toHaveBeenCalled()
    })
})
