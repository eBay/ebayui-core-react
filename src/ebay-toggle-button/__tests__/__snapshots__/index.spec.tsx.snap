// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EbayToggleButton /> should render component - <Default /> 1`] = `
<DocumentFragment>
  <button
    aria-pressed="false"
    class="toggle-button toggle-button--minimal-layout"
  >
    <div
      class="toggle-button__content"
    >
      <p
        class="toggle-button__title"
      >
        Button 1
      </p>
      <p
        class="toggle-button__subtitle"
      >
        subtitle
      </p>
    </div>
  </button>
</DocumentFragment>
`;

exports[`<EbayToggleButton /> should render component - <IconButton /> 1`] = `
<DocumentFragment>
  <button
    aria-pressed="false"
    class="toggle-button toggle-button--minimal-layout"
  >
    <div
      class="toggle-button__icon"
    >
      <svg
        aria-hidden="true"
        class="icon icon--24"
        focusable="false"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-settings-24"
        />
      </svg>
    </div>
    <div
      class="toggle-button__content"
    >
      <p
        class="toggle-button__title"
      >
        Button 1
      </p>
      <p
        class="toggle-button__subtitle"
      >
        Some context here
      </p>
    </div>
  </button>
</DocumentFragment>
`;

exports[`<EbayToggleButton /> should render component - <ImageButton /> 1`] = `
<DocumentFragment>
  <button
    aria-pressed="false"
    class="toggle-button toggle-button--minimal-layout"
  >
    <span
      class="toggle-button__image-container"
    >
      <span
        class="toggle-button__image"
      >
        <img
          alt="image alt"
          src="https://cloudfront.slrlounge.com/wp-content/uploads/2012/07/01-SLRLounge-Holding-Standing-Wrong.jpg"
        />
      </span>
    </span>
    <div
      class="toggle-button__content"
    >
      <p
        class="toggle-button__title"
      >
        Button 1
      </p>
      <p
        class="toggle-button__subtitle"
      >
        Some context here
      </p>
    </div>
  </button>
</DocumentFragment>
`;

exports[`<EbayToggleButton /> should render component - <ImageButtonWithPlacement /> 1`] = `
<DocumentFragment>
  <button
    aria-pressed="false"
    class="toggle-button toggle-button--gallery-layout"
  >
    <span
      class="toggle-button__image-container"
    >
      <span
        aria-label="image alt"
        class="toggle-button__image"
        role="img"
        style="background-image: url(https://cloudfront.slrlounge.com/wp-content/uploads/2012/07/01-SLRLounge-Holding-Standing-Wrong.jpg); background-position: top; background-size: cover; background-repeat: no-repeat;"
      />
    </span>
    <div
      class="toggle-button__content"
    >
      <p
        class="toggle-button__title"
      >
        Button 1
      </p>
      <p
        class="toggle-button__subtitle"
      >
        Some context here
      </p>
    </div>
  </button>
</DocumentFragment>
`;

exports[`<EbayToggleButton /> should render component - <WithChildren /> 1`] = `
<DocumentFragment>
  <button
    aria-pressed="false"
    class="toggle-button toggle-button--minimal-layout"
  >
    <div
      class="toggle-button__content"
    >
      <p>
        Button 1
      </p>
    </div>
  </button>
</DocumentFragment>
`;
